templates/stamp/mongodb-ubuntu-install.sh

```diff
@@ -0,0 +1,413 @@
+		done
+		
+		# Print the current replica set configuration
+		mongo master -u $ADMIN_USER_NAME -p $ADMIN_USER_PASSWORD --host 127.0.0.1 --eval "printjson(rs.conf())"	
+		mongo master -u $ADMIN_USER_NAME -p $ADMIN_USER_PASSWORD --host 127.0.0.1 --eval "printjson(rs.status())"	
+	fi
+	
+	# Register an arbiter node with the replica set
+	if [ "$IS_ARBITER" = true ]; then
+	
+		# Work out the IP address of the last member node where we initiated a replica set
+		let "PRIMARY_MEMBER_INDEX=$INSTANCE_COUNT-1"
+		PRIMARY_MEMBER_HOST="${NODE_IP_PREFIX}${PRIMARY_MEMBER_INDEX}:${MONGODB_PORT}"
+		CURRENT_NODE_IPS=`ifconfig | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*' | grep -v '127.0.0.1'`
+		CURRENT_NODE_IP=${CURRENT_NODE_IPS[@]}
+
+		log "Adding an arbiter ${HOSTNAME} ($CURRENT_NODE_IP) node to the replica set $REPLICA_SET_NAME"
+		mongo master -u $ADMIN_USER_NAME -p $ADMIN_USER_PASSWORD --host $PRIMARY_MEMBER_HOST --eval "printjson(rs.addArb('${CURRENT_NODE_IP}'))"
+	fi
+}
+
+#############################################################################
+
+configure_mongodb()
+{
+    log "Configuring MongoDB"
+
+    # first setup the unit file
+    log "Setting up the unit file"
+    create_mongodb_unitfile
+
+    log "Executing core configuration"
+	mkdir -p "$MONGODB_DATA"
+	mkdir "$MONGODB_DATA/log"
+	mkdir "$MONGODB_DATA/db"
+	
+	chown -R mongodb:mongodb "$MONGODB_DATA/db"
+	chown -R mongodb:mongodb "$MONGODB_DATA/log"
+	chmod 755 "$MONGODB_DATA"
+	
+	mkdir /var/run/mongodb
+	touch /var/run/mongodb/mongod.pid
+	chmod 777 /var/run/mongodb/mongod.pid
+	
+    # setup the linking jump point
+    # TEMPORARY WORK AROUND - there is an error seen when mongodb bootstraps directly from the linked blob, it crashes with unknown error
+    # as an interim work around, we are pointing mongodb to the local file system which has a symlink to the blob
+    log "Initiating local jump point at for MongoDB at $MONGODB_DATA_INTERIM_ROOTPATH"
+    mkdir -p "$MONGODB_DATA_INTERIM_ROOTPATH"
+    chown -R mongodb:mongodb "$MONGODB_DATA_INTERIM_ROOTPATH"
+	chmod 755 "$MONGODB_DATA_INTERIM_ROOTPATH"
+
+    ln -s "$MONGODB_DATA/log" "$MONGODB_DATA_INTERIM_ROOTPATH/log"
+    ln -s "$MONGODB_DATA/db" "$MONGODB_DATA_INTERIM_ROOTPATH/db"
+
+	tee /etc/mongod.conf > /dev/null <<EOF
+systemLog:
+    destination: file
+    path: "$MONGODB_DATA_INTERIM_ROOTPATH/log/mongod.log"
+    quiet: true
+    logAppend: true
+processManagement:
+    fork: true
+    pidFilePath: "/var/run/mongodb/mongod.pid"
+net:
+    port: $MONGODB_PORT
+security:
+    #keyFile: ""
+    authorization: "disabled"
+storage:
+    dbPath: "$MONGODB_DATA_INTERIM_ROOTPATH/db"
+    directoryPerDB: true
+    journal:
+        enabled: $JOURNAL_ENABLED
+#replication:
+    #replSetName: "$REPLICA_SET_NAME"
+EOF
+
+	# Fixing an issue where the mongod will not start after reboot where when /run is tmpfs the /var/run/mongodb directory will be deleted at reboot
```

Comment:
is the issue mongodb vs mongod? If yes, one refers to the application and the other refers to the server process.